avocado/utils/cpu.py

```diff
@@ -37,13 +37,16 @@ def _list_matches(lst, pattern):
 
 def _get_cpu_info():
     """
-    Reads /proc/cpuinfo and returns a list of file lines
+    Returns info on the 1st CPU entry from /proc/cpuinfo as a list of lines
 
-    :returns: `list` of lines from /proc/cpuinfo file
+    :returns: `list` of lines 1st CPU entry from /proc/cpuinfo file
     :rtype: `list`
     """
-    with open('/proc/cpuinfo', 'r') as f:
-        cpuinfo = f.readlines()
+    cpuinfo = []
+    for line in open('/proc/cpuinfo').readlines():
+        if line == '\n':
+            break
+        cpuinfo.append(line)
```

Comment:
Well `get_cpu_info` can mean anything and in my view it is `get /proc/cpuinfo content`. Here you are changing it to `get info about the first cpu` which is quite different as people could have used this function to eg. get number of cpus, or to get other information about n-th cpu as not all architectures are symetrical. Last but not least at least power reports some generic platform information at the end of the file.

So I think changing this concept is quite dangerous and I'd rather avoid it...